rialto_airflow/publish/publication.py

```diff
@@ -48,95 +48,77 @@ def init_reports_data_schema() -> None:
     create_schema(RIALTO_REPORTS_DB_NAME, ReportsSchemaBase)
 
 
-def write_publications(snapshot) -> Path:
+def write_publications(snapshot) -> bool:
     """
-    Write a CSV where there is a row per unique DOI.
+    Write publications information to the reports publications table
     """
-    col_names = [
-        "doi",
-        "pub_year",
-        "apc",
-        "open_access",
-        "types",
-        "funders",
-        "federally_funded",
-        "academic_council_authored",
-        "faculty_authored",
-    ]
 
-    csv_path = get_csv_path(snapshot, google_drive_folder(), "publications.csv")
-
-    logging.info(f"started writing publications {csv_path}")
-
-    with csv_path.open("w") as output:
-        csv_output = DictWriter(output, fieldnames=col_names)
-        csv_output.writeheader()
-
-        with get_session(snapshot.database_name).begin() as select_session:
-            # This query joins the publication and funder tables
-            # Since we want one row per publication, and a publication can
-            # have multiple funders, the funder names, and the booleans
-            # associated with whether they are federal, are grouped together in
-            # a list using the jsonb_agg_strict function (the strict version
-            # drops null values). In order to use these aggregate functions we
-            # need to group by the Publication.id.
-
-            stmt = (
-                select(  # type: ignore
-                    Publication.doi,  # type: ignore
-                    Publication.pub_year,  # type: ignore
-                    Publication.apc,  # type: ignore
-                    Publication.open_access,
-                    Publication.dim_json["type"].label("dim_type"),
-                    Publication.openalex_json["type"].label("openalex_type"),
-                    Publication.wos_json["static_data"]["fullrecord_metadata"][
-                        "normalized_doctypes"
-                    ]["doctype"].label("wos_type"),
-                    func.jsonb_agg_strict(Author.academic_council).label(
-                        "academic_council"
-                    ),
-                    func.jsonb_agg_strict(Author.primary_role).label("primary_role"),
-                    func.jsonb_agg_strict(Funder.name).label("funders"),
-                    func.jsonb_agg_strict(Funder.federal).label("federal"),
-                )
-                .join(Author, Publication.authors)  # type: ignore
-                .join(Funder, Publication.funders, isouter=True)  # type: ignore
-                .group_by(Publication.id)
-                .execution_options(yield_per=10_000)
```

Comment:
note: a lot of this code had indentation changed now that the CSV writing loop is gone, hence the big diff -- it was not actually deleted or altered